common/src/main/java/net/neoforged/gradle/common/dependency/AbstractDependencyManagementObject.java

```diff
@@ -4,47 +4,64 @@
 import net.neoforged.gradle.dsl.common.dependency.DependencyManagementObject;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Dependency;
-import org.gradle.api.artifacts.ModuleDependency;
-import org.gradle.api.artifacts.ResolvedDependency;
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier;
+import org.gradle.api.artifacts.dsl.DependencyFactory;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.ProviderFactory;
 import org.gradle.api.specs.Spec;
 import org.gradle.api.specs.Specs;
 
-import java.util.regex.Pattern;
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.Objects;
 
-public class AbstractDependencyManagementObject implements DependencyManagementObject {
+@SuppressWarnings("UnstableApiUsage")
+public abstract class AbstractDependencyManagementObject implements DependencyManagementObject {
+    @Inject
+    protected abstract DependencyFactory getDependencyFactory();
 
-    protected final Project project;
-
-    public AbstractDependencyManagementObject(Project project) {
-        this.project = project;
-    }
-
-    protected static ArtifactIdentifier createArtifactIdentifier(final ResolvedDependency dependency) {
-        return new ArtifactIdentifier(dependency.getModuleGroup(), dependency.getModuleName(), dependency.getModuleVersion());
+    protected static ArtifactIdentifier createArtifactIdentifier(final ModuleComponentIdentifier identifier) {
+        return new ArtifactIdentifier(identifier.getModuleIdentifier().getGroup(), identifier.getModuleIdentifier().getName(), identifier.getVersion());
     }
 
-    protected static ArtifactIdentifier createArtifactIdentifier(final ModuleDependency dependency) {
+    protected static ArtifactIdentifier createArtifactIdentifier(final Dependency dependency) {
         return new ArtifactIdentifier(dependency.getGroup(), dependency.getName(), dependency.getVersion());
     }
 
+    @SuppressWarnings("unchecked")
     public Spec<? super ArtifactIdentifier> dependency(Object notation) {
```

Comment:
Gradle is moving in the direction of typesafe dependency creation methods instead of the generic Object one (see, for example, the new DependencyFactory API). Might be worth changing this to match, and/or exposing the dependency factory itself somehow instead (do this via a closure/action mayhaps? I'll think about it)